action.yml

```diff
@@ -1,35 +1,134 @@
 name: Setup Nox
-description: 'Prepares all python versions for nox'
+description: "Prepares all python versions for nox"
+inputs:
+  python-versions:
+    description: "comma-separated list of python versions to install"
+    required: true
+    default: "3.7, 3.8, 3.9, 3.10, pypy-3.7, pypy-3.8, pypy-3.9"
 branding:
   icon: package
   color: blue
 
 runs:
   using: composite
   steps:
+    - name: "Validate input"
+      id: validate-input
+      run: |
```

Comment:
I think this would definitely benefit from being a standalone script, and I'd ideally like to see a decent number of test cases to ensure it handles weird edges, there's plenty of opportunity in here to fail in an ugly way (e.g. the indexing)

Reply:
> I think this would definitely benefit from being a standalone script

done

> and I'd ideally like to see a decent number of test cases to ensure it handles weird edges, there's plenty of opportunity in here to fail in an ugly way (e.g. the indexing).

will do if we decide to go further on this.

> However if this is only required because of supporting the slightly nicer list syntax then I'd argue the better approach is to simply live with the string list syntax.

that's not for a slightly nicer list syntax, more on that later.